File: ProBook/UI/probook-ui/src/app/interfaces/collectionResponse.ts
import { Notebook } from "./notebook-interface";
import { User } from "./user-interface";

export interface CollectionResponse{
    id?:number;
    name?:string;
    description?:string;
    createdAt?:Date;
    userId?:number;
    user?:User;
    notebooks?:Notebook[];
}

File: ProBook/ProBook.Model/Model/CollectionResponse.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ProBook.Model.Model
{
    public class CollectionResponse
    {

        public string? Name { get; set; }

        public string? Description { get; set; }

        public DateTime? CreatedAt { get; set; }

        public int UserId { get; set; }

        public virtual User? User { get; set; } = null!;

        public List<Notebook>? Notebooks { get; set; }
    }
}
